experimental/Schemes/CartierDivisor.jl

```diff
@@ -220,6 +220,9 @@ end
 function weil_divisor(C::EffectiveCartierDivisor;
     is_prime::Bool=false # Indicate whether this divisor is already prime
   )
+  return WeilDivisor(ideal_sheaf(C), ZZ, check=false)
+
+  # TODO: See what we can recycle from the code below.
```

Comment:
@afkafkafk13 : You once implemented more specialized routines to extract the multiplicities. It seems that this was still a bottleneck, though. For the moment I switched back to the proof-of-concept implementation using composition series, but this is also very slow. Depending on your discussions with @simonbrandhorst this afternoon, we should overhaul your code and find a suitable binding for it again. For instance, one could specify a task here to directly produce a `WeilDivisor` with irreducible `components`.